modules/health_quest/app/services/health_quest/health_api/location/factory.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+module HealthQuest
+  module HealthApi
+    module Location
+      ##
+      # A service object for isolating dependencies from any implementing service or controller.
+      #
+      # @!attribute session_service
+      #   @return [HealthQuest::Lighthouse::Session]
+      # @!attribute user
+      #   @return [User]
+      # @!attribute map_query
+      #   @return [HealthApi::Location::MapQuery]
+      # @!attribute options_builder
+      #   @return [Shared::OptionsBuilder]
+      class Factory
+        attr_reader :session_service, :user, :map_query, :options_builder
+
+        ##
+        # Builds a HealthApi::Location::Factory instance from a given User
+        #
+        # @param user [User] the currently logged in user.
+        # @return [HealthApi::Location::Factory] an instance of this class
+        #
+        def self.manufacture(user)
+          new(user)
```

Comment:
Is this method necessary? I feel like you can just do HealthQuest::HealthApi::Location::Factory.new(current_user)

Reply:
It's not strictly necessary from a purely functional perspective. It's just a design preference and convention. :D